kubernetes/deployments_kubeclient.go

```diff
@@ -0,0 +1,1339 @@
+	buildconfigs, err := kc.getBuildConfigs(spaceName)
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+
+	// Get all applications in this space using BuildConfig names
+	apps := []*app.SimpleApp{}
+	for _, bc := range buildconfigs {
+		appn, err := kc.GetApplication(spaceName, bc)
+		if err != nil {
+			return nil, errs.WithStack(err)
+		}
+		apps = append(apps, appn)
+	}
+
+	result := &app.SimpleSpace{
+		Type: "space",
+		Attributes: &app.SimpleSpaceAttributes{
+			Name:         &spaceName,
+			Applications: apps,
+		},
+	}
+
+	return result, nil
+}
+
+// GetApplication retrieves an application with the given space and application names, with the status
+// of that application's deployment in each environment
+func (kc *kubeClient) GetApplication(spaceName string, appName string) (*app.SimpleApp, error) {
+	// Get all deployments of this app for each environment in this space
+	deployments := []*app.SimpleDeployment{}
+	for envName := range kc.envMap {
+		deployment, err := kc.GetDeployment(spaceName, appName, envName)
+		if err != nil {
+			return nil, errs.WithStack(err)
+		} else if deployment != nil {
+			deployments = append(deployments, deployment)
+		}
+	}
+
+	result := &app.SimpleApp{
+		Type: "application",
+		Attributes: &app.SimpleAppAttributes{
+			Name:        &appName,
+			Deployments: deployments,
+		},
+		ID: appName,
+	}
+	return result, nil
+}
+
+// ScaleDeployment adjusts the desired number of replicas for a specified application, returning the
+// previous number of desired replicas
+func (kc *kubeClient) ScaleDeployment(spaceName string, appName string, envName string, deployNumber int) (*int, error) {
+	envNS, err := kc.getEnvironmentNamespace(envName)
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+	// Look up the Scale for the DeploymentConfig corresponding to the application name in the provided environment
+	dcScaleURL := fmt.Sprintf("/oapi/v1/namespaces/%s/deploymentconfigs/%s/scale", envNS, appName)
+	scale, err := kc.getResource(dcScaleURL, true)
+	if err != nil {
+		return nil, errs.WithStack(err)
+	} else if scale == nil {
+		return nil, nil
+	}
+
+	spec, ok := scale["spec"].(map[interface{}]interface{})
+	if !ok {
+		return nil, errs.New("invalid deployment config returned from endpoint: missing 'spec'")
+	}
+
+	replicasYaml, pres := spec["replicas"]
+	oldReplicas := 0 // replicas property may be missing from spec if set to 0
+	if pres {
+		oldReplicas, ok = replicasYaml.(int)
+		if !ok {
+			return nil, errs.New("invalid deployment config returned from endpoint: 'replicas' is not an integer")
+		}
+	}
+	spec["replicas"] = deployNumber
+
+	yamlScale, err := yaml.Marshal(scale)
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+
+	_, err = kc.putResource(dcScaleURL, yamlScale)
+	if err != nil {
+		return nil, errs.WithStack(err)
+	}
+
+	return &oldReplicas, nil
+}
+
+func (kc *kubeClient) getConsoleURL(envNS string) (*string, error) {
+	path := "console/project/" + envNS
```

Comment:
you could use `fmt.Sprintf()` here

Reply:
But this is faster and more readable.

Reply:
@stooke it may be faster but I think we have different opinions on readability ;)

Reply:
Changed.

Reply:
ok